gremlin-go/driver/graphBinary.go

```diff
@@ -199,6 +201,90 @@ func mapReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return valMap, nil
 }
 
+func getSignedBytesFromBigInt(n *big.Int) []byte {
+	var one = big.NewInt(1)
+	switch n.Sign() {
+	case 0:
+		return []byte{}
+	case 1:
+		b := n.Bytes()
+		if b[0]&0x80 > 0 {
+			b = append([]byte{0}, b...)
+		}
+		return b
+	case -1:
+		length := uint(n.BitLen()/8+1) * 8
+		b := new(big.Int).Add(n, new(big.Int).Lsh(one, length)).Bytes()
+		if len(b) >= 2 && b[0] == 0xff && b[1]&0x80 != 0 {
+			b = b[1:]
+		}
+		return b
+	}
+	panic("unreachable")
+}
+
+func getBigIntFromSignedBytes(b []byte) *big.Int {
+	var newBigInt = big.NewInt(0).SetBytes(b)
+	var one = big.NewInt(1)
+	if len(b) == 0 {
+		return newBigInt
+	}
+	switch b[0] & 0x80 {
+	case 0x00:
+		newBigInt.SetBytes(b)
+		return newBigInt
+	case 0x80:
+		length := uint((len(b)*8)/8+1) * 8
+		b2 := new(big.Int).Sub(newBigInt, new(big.Int).Lsh(one, length)).Bytes()
+		b2 = b2[1:]
+		if b2[0] == 0x00 {
+			b2 = b2[1:]
+		}
+		newBigInt = big.NewInt(0)
+		newBigInt.SetBytes(b2)
+		newBigInt.Neg(newBigInt)
+		return newBigInt
+	}
+	panic("unreachable")
+}
+
+// Format: {length}{value_0}...{value_n}
+func bigIntWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	v := value.(*big.Int)
+	signedBytes := getSignedBytesFromBigInt(v)
+	err := binary.Write(buffer, binary.BigEndian, int32(len(signedBytes)))
+	if err != nil {
+		return nil, err
+	}
+	if len(signedBytes) < 1 {
+		return buffer.Bytes(), nil
+	}
+	for i := 0; i < len(signedBytes); i++ {
+		err := binary.Write(buffer, binary.BigEndian, signedBytes[i])
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I think you can remove the length < 1 check above, without it, you will just break out of your for loop without iterating, effectively skipping it and doing the same thing anyway.
```suggestion
	for i := 0; i < len(signedBytes); i++ {
		err := binary.Write(buffer, binary.BigEndian, signedBytes[i])
		if err != nil {
			return nil, err
		}
	}
```